### DISCUSSION TITLE
Raspberry pi 5 and Agopengps

### DISCUSSION THREAD

**Discussion by ronantaylorware** [geezacoleman/OpenWeedLocator#123] [General]
G'day,

This looks awesome.  Have just received a raspberry pi 5 and the raspberry pi global shutter camera to test this out. Seems to be plenty of stock available these days. Not super cheap but reasonable prices I think.  Want to detect green on brown to target sparse summer weeds, where we currently toss up between chipping them out by hand or spraying the whole paddock. Unsure whether we'll put this on an old 1000L tank with 3 section 12m boom or newer 4000L 4 section 20m spray cart. Will stick to pulling with tractor for testing. I have a few initial questions..

Picked up the raspberry global shutter camera rather than arducam as its in stock locally. Is fairly new and low resolution but hoping it's a good fit. Has anyone had this going?

Noticed the raspberry pi 5 has 2 camera ports. Has anyone got this going with 2 cameras per pi?

Could one camera detect 5m width? Any ideas on width limit?

We currently use open source agopengps for gps and section control, with similar looking 4 channel relay boards.  Will be looking at combining OWL with agopen for sure. Something to note - some spray control manifolds use electric gear motors instead on solenoids - in that case the small blue relays simply send a 5v signal to turn sections on and off. In other cases if you're controlling solenoids the little blue relays may burn out due to greater amp draw (I think) and you may need a contactor after the relay.

Would be interesting to have agopen send signals to raspberry pi running owl instead of an arduino... but plan to simply install OWL further down the line.  So spraying is controlled primarily by agopengps and whether sections have been overlapped, and lastly whether weeds are detected.

Anyway thought I'd put my thoughts out there. We will be busy sowing soon so no idea when I'll have time to get into this properly. Any ideas anyone can share would be great. Cheers.

Ronan

**Comment by geezacoleman** [comment]
Hi Ronan,

This is fantastic - looking forward to seeing how you progress with the sprayer set up and getting your feedback/ideas for the OWL. We're building a 12m one as well later this year.

>Picked up the raspberry global shutter camera rather than arducam as its in stock locally. Is fairly new and low resolution but hoping it's a good fit. Has anyone had this going?

Finally got this working in an relatively untested form today. It's over on the [picamera2 branch](https://github.com/geezacoleman/OpenWeedLocator/discussions/123) if you're interested. Just clone it and set it up as normal. It uses a config file in the config directory to set all the parameters though. Runs at up to 60 FPS at 416 x 320 resolution, or 30 FPS at 640 x 480. So we should get much better performance. I'll be merging this into the main branch in the comiung months once it's more complete.

> Noticed the raspberry pi 5 has 2 camera ports. Has anyone got this going with 2 cameras per pi?

I haven't heard anyone using the Pi 5 for two streams with the OWL, but feasible for sure. Depends on the use case I guess, but could save on the Pi's. It seems to be the way the weed detection companies are going.

> Could one camera detect 5m width? Any ideas on width limit?

Resolution is the biggest limitation - for large weeds it would work, but if you're limited to 640 x 480, you would start to miss the small and medium sized weeds. Could run multiple cameras like you suggest above though to save on the RPi's. Lens choice would be key too, to get it the right width at the right height.

Will be cool to see AgOpenGPS and OWL working on the same system if you get it set up, it was a big inspiration behind the approach to the OWL.

Keep us updated though and send any questions through if you run into any issues.

**Comment by david-j-webster** [comment]
Ronan
Looks like a good project can you keep me informed as i would like to use the pi 5 and multiple cameras, as well
